play25-json-ops/src/main/scala/play/api/libs/json/ops/JsonImplicits.scala

```diff
@@ -20,7 +20,8 @@ trait JsonImplicits extends ImplicitTupleFormats with JsValueImplicits {
    * @param readKey a function that converts String to K or throws an exception if the conversion cannot be made.
    * @param writeKey a function that converts K to a String.
    */
-  implicit def formatMap[K, V: Format](
+  @deprecated("This method is unsafe and will be removed (possibly replaced by a more sound implementation).", "2.0.0")
+  def formatMap[K, V: Format](
```

Comment:
I think this will end up being a breaking change anyway. We might want to just release 3.0.0 with the correct fix.

Reply:
If I'm gonna go with a 3.0.0 fix, then it probably makes sense to pull out a common `playX-json-ops-common` project with cross-compatible play-json code. Also, I should add support for Play 2.6 / Scala 2.12 and Play 2.7.0-M1 / Scala 2.13.0-M4